### PR TITLE
Do not close the ssl socket twice

### LINKED ISSUE #6040: Crash with "abort+XXX" when call  network.request often
**Describe the bug**
Crash with "abort+XXX" when call  network.request often.

**To Reproduce**
- Do lot of call `network.request` method

**Expected behavior:**
Application crashed with `abort+XXX`

**Defold version:**
  - Version 1.2.187 BETA (1.2.186)

**Platforms (REQUIRED):**
 - Platforms: Android
 - OS: Android 9,10,11
 - Various (from budget models to top devices)

**Logs:**
```
pid: 0, tid: 0 >>> com.xiart.fb.solitaire <<<
backtrace:
  #00  pc 0000000000062de8  /apex/com.android.runtime/lib/bionic/libc.so (abort+172)
  #00  pc 0000000000065b23  /apex/com.android.runtime/lib/bionic/libc.so (fdsan_error(char const*, ...)+426)
  #00  pc 0000000000065837  /apex/com.android.runtime/lib/bionic/libc.so (android_fdsan_close_with_tag+470)
  #00  pc 0000000000065eab  /apex/com.android.runtime/lib/bionic/libc.so (close+6)
  #00  pc 000000000020021c  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (mbedtls_net_free) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/mbedtls/library/net_sockets.c:665)
  #00  pc 000000000017f13c  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmSSLSocket::Delete(dmSSLSocket::SSLSocket*)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/sslsocket.cpp:361)
  #00  pc 0000000000195a38  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmConnectionPool::DoClose(dmConnectionPool::ConnectionPool*, dmConnectionPool::Connection*)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/connection_pool.cpp:216)
  #00  pc 0000000000195a38  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmConnectionPool::PurgeExpired(dmConnectionPool::ConnectionPool*)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/connection_pool.cpp:228)
  #00  pc 0000000000195a38  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmConnectionPool::DoDial(dmConnectionPool::ConnectionPool*, char const*, unsigned short, bool, int, int*, unsigned int*, dmSocket::Result*, bool, bool)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/connection_pool.cpp:365)
  #00  pc 0000000000195f34  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmConnectionPool::Dial(dmConnectionPool::ConnectionPool*, char const*, unsigned short, bool, int, int*, unsigned int*, dmSocket::Result*)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/connection_pool.cpp:407)
  #00  pc 00000000001667c8  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmHttpClient::Response::Connect(char const*, unsigned short, bool, int, int*)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/http_client.cpp:195)
  #00  pc 0000000000167110  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmHttpClient::DoRequest(dmHttpClient::Client*, char const*, char const*)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/http_client.cpp:998)
  #00  pc 000000000011be1c  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000)
  #00  pc 000000000011c100  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000)
  #00  pc 0000000000176fd0  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmMessage::InternalDispatch(unsigned long long, void (*)(dmMessage::Message*, void*), void*, bool)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/message.cpp:541)
  #00  pc 00000000001771e4  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmMessage::DispatchBlocking(unsigned long long, void (*)(dmMessage::Message*, void*), void*)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/message.cpp:573)
  #00  pc 000000000011c618  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmHttpService::Loop(void*)) (SourceCode: /home/runner/work/defold/defold/engine/script/build/../src/http_service.cpp:330)
  #00  pc 0000000000180454  /data/app/~~Bn4LzZeqCT0joIhLhLAybg==/com.xiart.fb.solitaire-3UO00kXtheMUHuHcSWI7Ig==/split_config.armeabi_v7a.apk!lib/armeabi-v7a/libSolitaireTowersTournaments.so (offset 0x1000) (dmThread::ThreadStartProxy(void*)) (SourceCode: /home/runner/work/defold/defold/engine/dlib/build/../src/dlib/thread.cpp:42)
  #00  pc 00000000000aaf33  /apex/com.android.runtime/lib/bionic/libc.so (__pthread_start(void*)+40)
  #00  pc 00000000000642e3  /apex/com.android.runtime/lib/bionic/libc.so (__start_thread+30)
```
**1.2.187 BETA**
image
image
image
image

**1.2.186**
image

### PR DESCRIPTION
**Pull request by britzl**
Fixes #6040

### CODE REVIEW

File: engine/dlib/src/dlib/connection_pool.cpp [hunk 1]
```diff
@@ -208,13 +208,13 @@ namespace dmConnectionPool
 
     static void DoClose(HPool pool, Connection* c)
     {
+        if (c->m_SSLSocket != dmSSLSocket::INVALID_SOCKET_HANDLE) {
```
Comment (britzl) [inline engine/dlib/src/dlib/connection_pool.cpp:211]: Order changed as discussed

Comment (JCash) [inline engine/dlib/src/dlib/connection_pool.cpp:211]: Vart sätts INVALID_SOCKET_HANDLE när man delete'ar sslsocketen?

File: engine/dlib/src/dlib/connection_pool.cpp [hunk 2]
```diff
@@ -208,13 +208,13 @@ namespace dmConnectionPool
...
+            dmSSLSocket::Delete(c->m_SSLSocket);
+        }
         if (c->m_Socket != dmSocket::INVALID_SOCKET_HANDLE) {
```
Comment (JCash) [inline engine/dlib/src/dlib/connection_pool.cpp:214]: Same question here, really, where does the INVALID_SOCKET_HANDLE get assigned?

**Review by JCash — DISMISSED** [review]

File: engine/dlib/src/dlib/connection_pool.cpp (hunk 1 above)
Comment (britzl) [reply engine/dlib/src/dlib/connection_pool.cpp:211]: It's set when c->Clear() is called:

https://github.com/defold/defold/blob/dev/engine/dlib/src/dlib/connection_pool.cpp#L43-L49

https://github.com/defold/defold/blob/dev/engine/dlib/src/dlib/connection_pool.cpp#L218

**Review by JCash — APPROVED** [review]

**Pull request merged**